packages/components/controlled-input/src/ControlledInput.tsx

```diff
@@ -114,22 +121,21 @@ export const ControlledInput = ({
         required={required}
         disabled={disabled}
         onKeyDown={handleKeyDown}
-        {...otherProps}
+        {...inputProps}
       />
       {type === 'radio' ? (
-        /* eslint-disable-next-line jsx-a11y/label-has-associated-control */
+        // eslint-disable-next-line jsx-a11y/label-has-associated-control
```

Comment:
I think we should nest inputs inside `<label />`, I understand that it has `for` attribute but apparently this is not sufficient enough for all screen readers, here is a nice friendly [thread](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/718) 😅 
I'd say we should go for both, unless someone has a really good reason not to

Reply:
The way I read that thread it's actually better to use ids and not nest?

Reply:
yeah, obviously there's a conflict people some say ID is better, and some say nesting. I suggest we go for both, unless there's a reason not to

Reply:
Thanks for the little journey Moe. I've read through the whole issue and the tweet incl. comments. I also had a look into all the linked references (W3, MDN, a11ysupport) and I also understand that it's better to stick to the standard (using the explicit form with `for`). Some even state that doing both will cause bugs so there is no evidence that we will gain anything from doing both. It's not explicitly stated at W3N/ MDN that you may do both. So I would even say that we break the standard by doing so.

Looking at [a11ysupport](https://a11ysupport.io/tech/html/label_element), the explicit form (`for`) has better support for screen readers as the implicit form (wrapping element)

Reply:
thanks for investigating, let's keep it as is then 👍🏼

Reply:
+1

> IMHO I prefer web standards over micro-optimizations for some not correctly functioning voice-over software that is not sticking to the standards.

That depends on the situation I'd say, let's be pragmatic. If the Dragon device is the most used screen reader then we'd be shooting ourselves in the foot by not adding a simply fix. However, to me, going with just ids seems to be fine here.